src/ResourceManager/DataLakeStore/Commands.DataLakeStore/DataPlaneModels/AdlsClientFactory.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using Microsoft.Azure.Commands.Common.Authentication;
+using Microsoft.Azure.Commands.Common.Authentication.Abstractions;
+using Microsoft.Azure.DataLake.Store;
+using System.Text.RegularExpressions;
+using Microsoft.Azure.DataLake.Store.MockAdlsFileSystem;
+using Microsoft.Rest;
+
+namespace Microsoft.Azure.Commands.DataLakeStore.Models
+{
+    public class AdlsClientFactory
+    {
+        private static readonly Dictionary<string, AdlsClient> ClientFactory=new Dictionary<string, AdlsClient>();
+        /// <summary>
+        /// For unit testing this is set as true
+        /// </summary>
+        public static bool IsTest = false;
+        /// <summary>
+        /// Mock client credentials used for testing, this is set for record
+        /// </summary>
+        public static ServiceClientCredentials MockCredentials = null;
+        private static string HandleAccntName(string accnt,IAzureContext context)
+        {
+            if (Regex.IsMatch(accnt, @"^[a-zA-Z0-9]+$"))
+            {
+                return $"{accnt}.{context.Environment.GetEndpoint(AzureEnvironment.Endpoint.AzureDataLakeStoreFileSystemEndpointSuffix)}";
+            }
+            return accnt;
+        }
+        internal static AdlsClient GetAdlsClient(string accntNm, IAzureContext context)
```

Comment:
This is going to make things mroe difficult, I believe.  Instead, please use the uthenticationFactory to get the corrext token and use ClientFactory.GetCusotomArmClient to create the client with the appropriate settings.

Reply:
@markcowl As discussed currently my client does not inherit from Microsoft.Rest.ServiceClient<TClient>  since it is a custom dataplane so I cannot use ClientFactory.CreateCusotomArmClient. Currently by design for performance reasons we do not use HttpClient but use HttpWebRequest. We will look into HttpClient for future. So currently we have used the above way to create our custom client.

Reply:
@markcowl  Issue filed: https://github.com/Azure/azure-powershell/issues/5071